src/rhsm/config.py

```diff
@@ -108,8 +108,27 @@ def in_container() -> bool:
     # off
     if os.environ.get("SMDEV_CONTAINER_OFF", ""):
         return False
+
+    def in_ocp() -> bool:
+        """
+        Is the system running as pod in OCP (OpenShift Container Platform)?
+
+        Check some of the canonical environment variables set by Kubernets
+        (on which OCP is based):
+        https://kubernetes.io/docs/concepts/containers/container-environment/
```

Comment:
This page does not mention any of following variables. I found this: https://kubernetes.io/docs/reference/kubectl/#in-cluster-authentication-and-namespace-overrides, but it looks like that environment variables `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` are internal thing. They could make decision to change it in the future.

I would rather see something explicitly described in the documentation, because then it is harder to change it.

Reply:
> This page does not mention any of following variables.

It mentions `FOO_SERVICE_HOST` & `FOO_SERVICE_PORT`, set for each service named "foo". In particular, there exists a service named "kubernetes" in the cluster:
```
$ kubectl describe svc kubernetes
Name:              kubernetes
Namespace:         default
Labels:            component=apiserver
                   provider=kubernetes
Annotations:       <none>
Selector:          <none>
Type:              ClusterIP
IP Family Policy:  SingleStack
IP Families:       IPv4
IP:                172.30.0.1
IPs:               172.30.0.1
Port:              https  443/TCP
TargetPort:        6443/TCP
Endpoints:         10.0.137.183:6443,10.0.188.106:6443,10.0.221.126:6443
Session Affinity:  None
Events:            <none>
```
Thus, `KUBERNETES_SERVICE_HOST` & `KUBERNETES_SERVICE_PORT` are set.

See also:
- https://stackoverflow.com/questions/47523136/whats-the-purpose-of-the-default-kubernetes-service
- https://discuss.kubernetes.io/t/what-is-the-use-of-kubernetes-service-in-default-namespace/17545
- https://networkop.co.uk/post/2020-06-kubernetes-default/

Reply:
Yeah, there probably isn't better documentation for that.